server/clone.go

```diff
@@ -65,9 +65,13 @@ func (s *cloneService) Clone(ctx context.Context, req *agentrpc.CloneRequest) (*
 	if !s.agent.sem.TryAcquire(1) {
 		return nil, status.Error(codes.ResourceExhausted, "another request is under processing")
 	}
+	metrics.SetCloneInProgressMetrics(s.agent.clusterName, true)
```

Comment:
This also can be postponed.
Just before `clone()`, or inside of `clone()` as other metrics.